dulwich/tests/test_porcelain.py

```diff
@@ -1705,32 +1720,66 @@ def test_get_untracked_paths_nested(self):
             ),
         )
         self.assertEqual(
+            set(),
             set(
-                [
-                    os.path.join("nested", "ignored"),
-                    os.path.join("nested", "with"),
-                    os.path.join("nested", "manager"),
-                ]
+                porcelain.get_untracked_paths(
+                    subrepo.path,
+                    self.repo.path,
+                    self.repo.open_index(),
+                )
             ),
+        )
+        self.assertEqual(
+            set([os.path.join('nested', 'ignored'),
+                os.path.join('nested', 'with'),
+                os.path.join('nested', 'manager')]),
             set(
                 porcelain.get_untracked_paths(
                     self.repo.path,
                     subrepo.path,
                     self.repo.open_index(),
-                    exclude_ignored=False,
                 )
             ),
         )
+
+    def test_get_untracked_paths_subdir(self):
+        with open(os.path.join(self.repo.path, ".gitignore"), "w") as f:
+            f.write("subdir/\nignored")
+        with open(os.path.join(self.repo.path, "notignored"), "w") as f:
+            f.write("blah\n")
+        os.mkdir(os.path.join(self.repo.path, "subdir"))
+        with open(os.path.join(self.repo.path, "ignored"), "w") as f:
+            f.write("foo")
+        with open(os.path.join(self.repo.path, "subdir", "ignored"), "w") as f:
+            f.write("foo")
+
+        self.assertEqual(
+            set(
+                [
+                    ".gitignore",
+                    "notignored",
+                    "ignored",
+                    os.path.join("subdir", ""),
+                ]
+            ),
```

Comment:
This test should catch any future regression, if we recurse into the ignored `subdir` here, we will end up yielding `subdir/ignored` as well.